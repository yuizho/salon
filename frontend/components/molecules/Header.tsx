import { FC, useState } from 'react';
import ErrorAlert from '../atoms/ErrorAlert';
import NavBar from '../atoms/NavBar';

type ComponentProps = {
  message: string;
  error: boolean;
  onClose: () => void;
};

export const Component: FC<ComponentProps> = ({ message, error, onClose }) => (
  <>
    <NavBar />
    <ErrorAlert message={message} error={error} onClose={onClose} />
  </>
);

const Container: FC = () => {
  // TODO: use recoil states
  const [error, setError] = useState(false);
  const [message] = useState('error hoge');

  return <Component message={message} error={error} onClose={() => setError(false)} />;
};

export default Container;
