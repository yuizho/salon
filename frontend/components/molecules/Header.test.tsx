import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import Header from './Header';

describe('Header', () => {
  test('snapshot', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <Header />
      </RecoilRoot>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
