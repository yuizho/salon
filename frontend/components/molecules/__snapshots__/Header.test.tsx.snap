// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex items-center justify-between flex-wrap p-4 border-b"
  >
    <div
      class="flex items-center flex-shrink-0 text-slate-800"
    >
      <span
        class="font-semibold text-2xl tracking-tight"
      >
        Salon
      </span>
    </div>
  </nav>
  <div />
</DocumentFragment>
`;
